homeassistant/components/light/child_light.py

```diff
@@ -0,0 +1,195 @@
+"""
+homeassistant.components.light.child_light
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+his component is designed to enable other components to create a light
+without the nned to create an entirely new light component.
+The child light derives its state from monitoring one of the parent
+component's state attributes and will call a specified service when
+toggled.
+
+BASIC USAGE:
+
+You should expose the functionality you want the light to control in your
+parent component.  The child lights are created using HA's discovery
+mechanism.  You simply need to define the state attribute to be montiored
+and the service to be called on toggle and fire the discovery event from
+your component.
+
+from homeassistant.components.light import (
+    DISCOVER_CHILD_LIGHTS)
+
+from homeassistant.const import (
+    EVENT_PLATFORM_DISCOVERED)
+
+...
+
+# In your 'setup_platform' register a service
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+
+    # setup up your component here
+
+    ...
+
+    # Register a service
+    def toggle_something_service(service):
+        my_custom_info = service.data.get('extra_data')
+
+        # Do stuff
+
+    hass.services.register(DOMAIN, MY_SERVICE_NAME, toggle_something_service)
+
+# In your 'setup_platform' or from inside the component do the following:
+
+data = {}
+data['name'] = 'Arm my sensor'
+data['parent_entity_id'] = self.entity_id
+data['watched_variable'] = 'brightness'
+data['parent_domain'] = DOMAIN
+data['parent_service'] = MY_SERVICE_NAME
+data['parent_action'] = 'arbitrary_action_name'
+data['state_attributes'] = {}
+data['light_type'] = 'dimmer'
+data['extra_data'] = {
```

Comment:
This attribute should be unnecessary. If the parent wants to store extra data, it can do so by holding onto the child entity id and have that passed back into the service.